content/03-type-system.md

```diff
@@ -294,15 +294,15 @@ Unification of types having or being a [monomorph](types-monomorph) is detailed
 <!--label:type-system-unification-function-return-->
 #### Function Return
 
-Unification of function return types may involve the [`Void`-type](types-void) and requires a clear definition of what unifies with `Void`. With `Void` describing the absence of a type, it is not assignable to any other type, not even `Dynamic`. This means that if a function is explicitly declared as returning `Dynamic`, it cannot return `Void`.
+Unification of function return types may involve the [`Void`](types-void) type and requires a clear definition of what unifies with `Void`. With `Void` describing the absence of a type, it is not assignable to any other type, not even `Dynamic`. This means that if a function is explicitly declared as returning `Dynamic`, it cannot return `Void`.
 
-The opposite applies as well: If a function declares a return type of `Void`, it cannot return `Dynamic` or any other type. However, this direction of unification is allowed when assigning function types:
+The opposite applies as well: if a function declares a return type of `Void`, it cannot return `Dynamic` or any other type. However, this direction of unification is allowed when assigning function types:
 
 ```haxe
 var func:Void->Void = function() return "foo";
 ```
 
-The right-hand function clearly is of type `Void->String`, yet we can assign it to the variable `func` of type `Void->Void`. This is because the compiler can safely assume that the return type is irrelevant, given that it could not be assigned to any non-`Void` type.
+The right-hand function is clearly of type `Void->String`, yet we can assign it to the variable `func` of type `Void->Void`. This is because the compiler can safely assume that the return type is irrelevant, given that it could not be assigned to any non-`Void` type.
```

Comment:
Maybe we should use the new function type syntax here. `()->String` and `()->Void`

Reply:
Though I think the convention for the new syntax is usually to have spaces around `->`?

Reply:
Is it? I don't like spaces around `->` :)

Reply:
I don't like the spaces either, but it is arguably clearer and already in the manual, see https://haxe.org/manual/types-function.html

Reply:
It's also consistent with how arrow functions are usually formatted. Or do you not put spaces there either?

Reply:
I put spaces when writing an arrow function, but not when writing a function type.

Reply:
Right, and the new function type syntax is very similar to / inspired by arrow functions, so it feels strange to me to use different styles for them.

Reply:
I'll leave this as is for now in this pull request and make a more global change (or not) once everyone's settled on how this should be handled. I think I'll just open an issue for discussion on general style stuff soon.